### DISCUSSION TITLE
Xiaomi End Device Sensors: Don't want to work with Hue Routers

### DISCUSSION THREAD

**Discussion by Fruchuxs** [Koenkk/zigbee2mqtt#6078] [Devices]
## What happened
Hey, I'm new with zigbee2mqtt. For all my lights (HUE, Dresden Elektronik, one Iluminize Relay and one Osram Smart Plug and lots of Xiaomi Sensors) it works perfectly. I use the slae.sh CC2652RB stick.
But with my Xiaomi Devices, I have lots of problems. I readed [this](https://community.hubitat.com/t/xiaomi-aqara-devices-pairing-keeping-them-connected/623), and [this ](https://community.smartthings.com/t/original-aqara-xiaomi-zigbee-sensors-contact-temp-motion-button-outlet-leak-etc/113253) and a lot more stuff, but nothing works.

Can I pair the devices? Yes, I can if the devices near the coordinator, but If i move the sensors into the rooms where they should be, I got problems. In the linked articles above, there is mentioned, that I should pair the devices at the final location, but this doesnt work.

Maybe look at this:
![image](https://user-images.githubusercontent.com/5205672/106809852-77f55500-666c-11eb-81e1-1179c8470242.png)

Red = The coordinator
Green = The "not working" devices.

So what have I done:
- Tried to pair it in the final location: long press until flashing, wait. Just one "blink" every time. Tried multiples times. I also tried to keep pressing the button so the devices wont sleep. In the bottom left room this worked, but the devices dropped the connection after nine hours.
- Replaced the battery with a new one
- All Lamps u can see at the picture are Philips Hue Lamps
- Tried to disconnect the smart plug in the tp middle room, because I read they did problems with xiaomi
- Pair it near the coordinator and moved to its final location; connection was dropped after a hour

Maybe it's interesting to note that I moved my devices from Deconz/RaspBeeII to the new  Stick and mqtt because my light control with deconz was just bad (rly slow, the RaspBee was busy - no problems with zb2mqtt), but deconz never had problems with the xiaomi sensors, it's kinda joke, that I solved one problem with the hardware/software switch, but I got another problem.. ._.

## What did you expect to happen
I expect that the Xiaomi Devices paired into the network through one of my philips hue lamps which should act as a router (yes, permit join is true).

## How to reproduce it (minimal and precise)
Try to connect a Xiaomi Device (contact sensor, Multi Sensor and Presence Sensor) away from the coordinator sourrounded by rly many routers.

## Debug info
Zigbee2MQTT version: 1.17.1 (upgraded yesterday)
Adapter hardware:  CC2652RB 
Extension Cables: Yes, no USB 3 Devices connected or in range; just a HomeaticIP Dongle
Adapter firmware version: 20210120 (upgraded today)
System: RPI4

**Comment by Fruchuxs** [comment]
Thank you for moving it. But I created my post with some effort to be detailed as possible and I'm happy If you can explain me, why this is a discussion topic and not a problem?

**Reply by Koenkk** [reply]
Because this is probably not directly a Z2M issue, if certain devices cannot be paired through others we cannot do anything about that from the Z2M side. If you want to debug this further you have to make a sniff and share it here together with your network key: https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

**Reply by Fruchuxs** [reply]
Yes, you are right. I thought about it, but i'm confused because the same setup worked fine with deconz/phoscon and is now broken with Z2M. I tried many things and atm i'm a little bit helpless. Will try the sniff, thank you.

**Reply by Fruchuxs** [reply]
Maybe it's also important to note, that I paired all my devices, created a Network Key and changed the channel (I forget this first) and then repaired the devices again. Maybe this is the root of my problems.

**Reply by Fruchuxs** [reply]
It is save to flash my stick with the sniffer software, or it is better to buy a separate stick?

I have a strange theory: Sometimes the Xiaomi Sensors pair with my illuminize relay (if I repair it). But the connection with the illuminize relay seems to be dropped after some hours. With your nice network map in the Z2M Web backend, I figured out, that the lamps who are not work with the sensors, are also connected with the relay. I assume the relay just drop some informations at the pairing process?

**Reply by Fruchuxs** [reply]
NVM, this is the same problem: https://github.com/Koenkk/zigbee2mqtt/issues/3941

Maybe I remove the illuminaze one for a while and repair the xiaomis ..

**Reply by Fruchuxs** [reply]
Okay, this worked fine. The Network repaired itself after removed the illuminize. Now the illuminize is unable to pair with Z2M. Rly frustating.

**Reply by RoboMagus** [reply]
Hey @Fruchuxs, have you found a solution to your problem by now?

I came across this topic just after I re-paired my whole network after changing coordinators and switching up some settings in the process. I noticed that not all devices seem to like all channels equally well. For me pairing went much more smoothly on channel 20.

Maybe you could try a couple of different channels to see what works in your situation?

**Reply by Fruchuxs** [reply]
Yep, to remove the illuminize router was the clue.
